pkg/client/operatorclient.go

```diff
@@ -80,7 +80,7 @@ func (c *OperatorHelperClient) UpdateOperatorStatus(ctx context.Context, resourc
 	updated.ResourceVersion = resourceVersion
 	updated.Status.OperatorStatus = *status
 
-	ret, err := c.client.Update(ctx, updated, metav1.UpdateOptions{})
+	ret, err := c.client.UpdateStatus(ctx, updated, metav1.UpdateOptions{})
```

Comment:
does `UpdateStatus` let us update `ResourceVersion`?

(and/or, is this implementation of `UpdateOperatorStatus` actually correct according to the current library-go? `v1helpers.UpdateStatus` has this whole thing about listers vs live GET, which seems to be thwarted by our using the lister only here?)

Reply:
@danwinship I think it should be fine. `ResourceVersion` should be used by the apiserver to detect the latest observed RV by the client to prevent conflicts. And from what I see we try to fetch the latest RV as a client and pass it. This field should be updatable by the apiserver only anyway.

Nevertheless, I have reverted the change to see if I missed something. Let's see if there is any change in the observed test failures from https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-network-operator/2506/pull-ci-openshift-cluster-network-operator-master-e2e-aws-ovn-serial/1836848497182117888

Reply:
did not help, still got the failures: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-network-operator/2506/pull-ci-openshift-cluster-network-operator-master-e2e-aws-ovn-serial/1838337134189088768

Reply:
testing without any CNO changes now...

Reply:
after removing the ApplyStatus, the job has passed :shrug:, I am not sure where is the problem  https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-network-operator/2506/pull-ci-openshift-cluster-network-operator-master-e2e-aws-ovn-serial/1838512285375533056 

anyway, retesting the other jobs, as they do not seem to indicate they are caused by this change

Reply:
So we need the changes, as we can see unnecessary status reconciliation in the logs https://github.com/openshift/cluster-network-operator/pull/2506#issuecomment-2383308291. I have added the UpdateStatus/ApplyStatus back. I have also tested the failing tests in my cluster and could not reproduce the failures. Let's see is we observe the failiures again.